packages/assistant-stream/src/core/serialization/data-stream/DataStream.ts

```
@@ -231,6 +248,17 @@ export class DataStreamDecoder extends PipeableTransformStream<
               break;
             }
 
+            case DataStreamStreamChunkType.AuiToolCallArtifact: {
+              const { toolCallId, artifact } = value;
+              const toolCallController = toolCallControllers.get(toolCallId);
+              if (!toolCallController)
+                throw new Error(
+                  "Encountered tool call with unknown id: " + toolCallId,
```

Biome (1.9.4)

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)